core/src/main/java/org/mule/runtime/core/internal/util/generics/GenericTypeResolver.java

```diff
@@ -196,7 +196,7 @@ static Type getRawType(Type genericType, Map<TypeVariable, Type> typeVariableMap
     Map<TypeVariable, Type> typeVariableMap = (ref != null ? ref.get() : null);
 
     if (typeVariableMap == null) {
-      typeVariableMap = new HashMap<TypeVariable, Type>();
+      typeVariableMap = new HashMap<>();
```

Comment:
why not this one?

Reply:
because in this case the keys' equals() is not faster than its hashCde

Reply:
d'oh!